frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -91,32 +109,48 @@ export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps)
       <ModalBody>
         Adding capacity for <strong>{getName(ocsConfig)}</strong>, may increase your cloud expenses.
         <div className="ceph-add-capacity__modal">
-          <div className="ceph-add-capacity_sc-dropdown">
-            <OCSStorageClassDropdown onChange={setStorageClass} defaultClass={storageClass} />
+          <div
+            className={`ceph-add-capacity_sc-dropdown ${
+              isNoProvisionerSC ? '' : 'ceph-add-capacity_sc-dropdown__margin'
+            }`}
+          >
+            <OCSStorageClassDropdown onChange={handleStorageClass} defaultClass={storageClass} />
```

Comment:
Remove this. Use the Original storage class dropdown which will give you the whole `StorageClass` object. Perform all the operations with it. There is no point in storing the provisioner information and storage class name in different states. It just makes the thing more complicated than required.

Reply:
This `OCSStorageClassDropdown` filters out the `cephStorageProvisioners`and its used at two places. That was whole idea behind creating this component. Also when there will be support for baremetal infra, changes need to be done here only. Better to create a component for it and used it.

Reply:
Sure, my concern is the use of two states to store information related storageClass which should be in one state. Such a design is error-prone. When one is updated the other should be updated.(can lead to bugs). I would prefer a single source of truth. Maybe that can be handled with the same component ( if it returns a storage class resource).

Reply:
As I mentioned, we need this separate component since we are filtering and doing manipulations which is not present in the original component. Hence, we need this. We have tried to use the original one, it doesn't fulfill the functionalities.

Reply:
As I have clarified above my concern is two states to store information that should stay together.